Crash report 4.5.1: NPE in StatsAbstractInsightsFragment.showErrorUI

```
de.greenrobot.event.EventBusException: Invoking subscriber failed
       at de.greenrobot.event.EventBus.handleSubscriberException(EventBus.java:518)
       at de.greenrobot.event.EventBus.invokeSubscriber(EventBus.java:500)
       at de.greenrobot.event.EventBus.invokeSubscriber(EventBus.java:492)
       at de.greenrobot.event.HandlerPoster.handleMessage(HandlerPoster.java:67)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:135)
       at android.app.ActivityThread.main(ActivityThread.java:5254)
       at java.lang.reflect.Method.invoke(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:372)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:903)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:698)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.String.contains(java.lang.CharSequence)' on a null object reference
       at org.wordpress.android.ui.stats.StatsAbstractInsightsFragment.showErrorUI(StatsAbstractInsightsFragment.java:169)
       at org.wordpress.android.ui.stats.StatsInsightsLatestPostSummaryFragment.onEventMainThread(StatsInsightsLatestPostSummaryFragment.java:44)
       at java.lang.reflect.Method.invoke(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:372)
       at de.greenrobot.event.EventBus.invokeSubscriber(EventBus.java:498)
       at de.greenrobot.event.EventBus.invokeSubscriber(EventBus.java:492)
       at de.greenrobot.event.HandlerPoster.handleMessage(HandlerPoster.java:67)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:135)
       at android.app.ActivityThread.main(ActivityThread.java:5254)
       at java.lang.reflect.Method.invoke(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:372)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:903)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:698)
```

--- a/WordPress/src/main/java/org/wordpress/android/ui/stats/StatsAbstractInsightsFragment.java
+++ b/WordPress/src/main/java/org/wordpress/android/ui/stats/StatsAbstractInsightsFragment.java
@@ -166,6 +166,11 @@ public abstract class StatsAbstractInsightsFragment extends StatsAbstractFragmen
             return;
         }
 
+        // Use the generic error message when the string passed to this method is null.
+        if (label == null) {
+            label = "<b>" + getString(R.string.error_refresh_stats) + "</b>";
+        }
+
         if (label.contains("<")) {
             mErrorLabel.setText(Html.fromHtml(label));
         } else {

--- a/WordPress/src/main/java/org/wordpress/android/ui/stats/StatsInsightsLatestPostSummaryFragment.java
+++ b/WordPress/src/main/java/org/wordpress/android/ui/stats/StatsInsightsLatestPostSummaryFragment.java
@@ -39,15 +39,17 @@ public class StatsInsightsLatestPostSummaryFragment extends StatsAbstractInsight
         if (event.mEndPointName != StatsService.StatsEndpointsEnum.INSIGHTS_LATEST_POST_VIEWS) {
             super.onEventMainThread(event);
         } else {
-            // Another additional check. It ensures that the main data is available, and it should be at this point!!
-            if (isDataEmpty(0) || !(mDatamodels[0] instanceof InsightsLatestPostModel)) {
-                showErrorUI(null);
-                return;
-            }
 
+            // Check the response of the 2nd rest call before going deeper into updating the UI.
             if (event.mResponseObjectModel instanceof VolleyError ||
                     event.mResponseObjectModel instanceof StatsError) {
                 showErrorUI(event.mResponseObjectModel);
+                return;
+            }
+
+            // Here an another additional check that ensures the main data returned from the 1st call is available.
+            if (isDataEmpty(0) || !(mDatamodels[0] instanceof InsightsLatestPostModel)) {
+                showErrorUI(null);
                 return;
             }
 

